### DISCUSSION TITLE
New "Bypass" for Hub2K

### DISCUSSION THREAD

**Discussion by devninja-sudo** [reinhard-brandstaedter/solarflow-control#262] [Ideas]
Hello everyone,
I have a Hub 2000 with a 2kWp solar system in my setup. As I don't use much energy during the day, the battery (4kWh) is charged, but this is only possible with ~1400W. As the consumption is very low, the rest of the energy is wasted on sunny days. My idea at this point would be a control system that only looks at Hub2K to see whether the battery is being charged with ~1400W and then calculates solar input - output pack power and controls the inverter accordingly.

I'll try to write such a function, but I'm not that familiar with Python. 

Thanks in advance

**Comment by reinhard-brandstaedter** [comment]
You can just use sf-control for that already. Just set the min_charge_level to 1400W and only excess will be fed to house.
Though I don't think thats the best way to use your 2kWp because you can easily charge at a slower speed and still get your 4kWh battery from 0-100% within a day.
